### PR TITLE
Make the default route for private subnets a var.

### PR DESCRIPTION
**Pull request by dogfish182**
Use case is in an appliance based enterprise world, you'll want/need to route traffic through something other than a NAT gateway for egress scanning.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -155,12 +155,19 @@ resource "aws_route_table" "private" {
...
+  destination_cidr_block = "0.0.0.0/0"
   nat_gateway_id         = aws_nat_gateway.default[count.index].id
 }
 
+resource "aws_route" "endpoint_egress_routing" {
+  count                  = var.enable_endpoint_egress_routing && local.private_subnets > 0 ? length(var.endpoint_egress_data) : 0
+  route_table_id         = var.endpoint_egress_data[count.index].route_table_id
+  destination_cidr_block = var.endpoint_egress_data[count.index].destination_cidr_block
+  vpc_endpoint_id        = var.endpoint_egress_data[count.index].vpc_endpoint_id
+}
```
Comment (svanharmelen) [inline main.tf:169]: Wouldn't it be easier to not include this in this module? I would suggest to _only_ make the default route optional. Any other customer specific rules can be added outside of this module in the customer environment.

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (dogfish182) [reply main.tf:169]: makes sense, I'll refactor to optionalize default route.

File: main.tf [hunk 2]
```diff
@@ -155,7 +155,7 @@ resource "aws_route_table" "private" {
 }
 
 resource "aws_route" "private" {
-  count                  = var.enable_nat_gateway && local.private_subnets > 0 && local.public_subnets > 0 ? local.private_subnets : 0
+  count                  = enable_private_default_route && var.enable_nat_gateway && local.private_subnets > 0 && local.public_subnets > 0 ? local.private_subnets : 0
```
Comment (svanharmelen) [inline main.tf:158]:
```suggestion
  count                  = var.enable_private_default_route && var.enable_nat_gateway && local.private_subnets > 0 && local.public_subnets > 0 ? local.private_subnets : 0
```

File: variables.tf [hunk 3]
```diff
@@ -66,6 +66,12 @@ variable "enable_nat_gateway" {
   description = "Set to true to provision a NAT Gateway for each private subnet"
 }
 
+variable "enable_private_default_route" {
+  type        = bool
+  default     = true
+  description = "Set to true to provision a default route towards NAT gateway for private subnets"
```
Comment (svanharmelen) [inline variables.tf:72]:
```suggestion
  description = "Set to true to add a default route to the NAT gateway for each private subnet"
```

**Review by svanharmelen — APPROVED** [review]

**Pull request merged**